Integrations/integration-DuoAdminApi.yml

```
@@ -0,0 +1,372 @@
+        '10_minutes_ago' : datetime.datetime.now() - datetime.timedelta(minutes = 10),
+        '1_hour_ago' : datetime.datetime.now() - datetime.timedelta(hours = 1),
+        '10_hours_ago' : datetime.datetime.now() - datetime.timedelta(hours = 10),
+        '1_day_ago' : datetime.datetime.now() - datetime.timedelta(days = 1),
+        '1_week_ago' : datetime.datetime.now() - datetime.timedelta(days = 7),
+        '1_month_ago' : datetime.datetime.now() - datetime.timedelta(days = 30),
+        '1_year_ago' : datetime.datetime.now() - datetime.timedelta(days = 365),
+        '5_years_ago' : datetime.datetime.now() - datetime.timedelta(days = 1825),
+        '10_years_ago' : datetime.datetime.now() - datetime.timedelta(days = 3650)
+
+    }
+
+    #Utility Methods
+
+    def time_to_timestamp_milliseconds(time):
+        return str(calendar.timegm(time.utctimetuple()) * 1000)
+
+    # Generic function that receives a result json, and turns it into an entryObject
+    def get_entry_for_object(title, obj, contents, context, headers=None, humanReadable=None):
+        if len(obj) == 0:
+            return {
+                'Type': entryTypes['note'],
+                'Contents': contents,
+                'ContentsFormat': formats['json'],
+                'HumanReadable': "There is no output result",
+                'EntryContext': context
+                }
+        if headers:
+            if isinstance(headers, basestring):
+                headers = headers.split(',')
+            if isinstance(obj, dict):
+                headers = list(set(headers).intersection(set(obj.keys())))
+        return {
+            'Type': entryTypes['note'],
+            'Contents': obj,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable':  tableToMarkdown(title, obj, headers, lambda h: h.title().replace("_"," ").replace(".",":")),
+            'EntryContext': context
+        }
+
+    def get_user_id(username):
+        res = admin_api.get_users_by_name(username)
+        if len(res) == 0:
+            return_error("No users found with the given username")
+
+        return res[0]['user_id']
+
+    #Methods
+
+    #Duo client return 2 different known structures of error messages
+    def test_instance():
+        try:
+            res = admin_api.get_info_summary()
+            demisto.results('ok')
+        except Exception as e:
+            if(hasattr(e,'data')):
+                #error data for 40103 is not informative enough so we write our own
+                if(e.data['code'] == 40103):
+                    demisto.results('Invalid secret key in request credentials')
+                else:
+                    demisto.results(e.data['message'])
+            elif(hasattr(e,'strerror')):
+                demisto.results(e.strerror)
+            else:
+                demisto.results('Unknown error')
```

add some info about the error in this case
```suggestion
                demisto.results('Unknown error: ' + str(e))
```